test/jdk/java/awt/Frame/FrameResizeTest/FrameResizeTest_2.java

```diff
@@ -0,0 +1,144 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Canvas;
+import java.awt.Color;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Panel;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @bug 4065568
+ * @key headful
+ * @summary Test resizing a frame containing a canvas
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual FrameResizeTest_2
+ */
+
+public class FrameResizeTest_2 {
+    private static final String INSTRUCTIONS = """
+        There is a frame (size 300x300).
+        The left half is red and the right half is blue.
+
+        When you resize the frame, it should still have a red left half
+        and a blue right half.
+
+        In particular, no green should be visible after a resize.
+
+        Upon test completion, click Pass or Fail appropriately.
+        """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("FrameResizeTest_2 Instructions")
+                .instructions(INSTRUCTIONS)
+                .testTimeOut(5)
+                .rows(10)
+                .columns(45)
+                .build();
+
+        SwingUtilities.invokeAndWait(() -> {
+            FrameResize_2 frame = new FrameResize_2();
+
+            PassFailJFrame.addTestWindow(frame);
+            PassFailJFrame.positionTestWindow(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+
+            frame.setVisible(true);
+        });
+
+        passFailJFrame.awaitAndCheck();
+    }
+}
+
+class FrameResize_2 extends Frame {
+
+    FrameResize_2() {
+        super("FrameResize_2");
+
+        setLayout( new GridBagLayout() );
+
+        GridBagConstraints c = new GridBagConstraints();
+        c.fill = GridBagConstraints.BOTH;
+        c.weightx = 1;
+        c.weighty = 1;
+
+        Container dumbc = new DumbC();
+        add( dumbc, c );
+
+        Panel dump = new DumbPanel();
+        add( dump, c );
+
+        setSize( new Dimension( 300, 300));
+    }
+}
+
+
+class Fake extends Canvas {
+    public Fake(String name, Color what) {
+        setBackground(what);
+        setName(name);
+    }
+
+    public void paint(Graphics g) {
+        Dimension d = getSize();
+        g.setColor(getBackground());
+        g.fillRect(0, 0, d.width, d.height);
+    }
+}
+
+class DumbC extends Container {
```

Comment:
```suggestion
class DumbContainer extends Container {
```
Similarly, to `DumbPanel`?